pkg/steps/cluster_claim.go

```diff
@@ -61,7 +61,33 @@ func (s *clusterClaimStep) run(ctx context.Context) error {
 		// should never happen
 		return fmt.Errorf("cannot claim a nil cluster")
 	}
-	clusterClaim, err := acquireCluster(ctx, *s.clusterClaim, s.hiveClient, s.client, *s.jobSpec)
+	waitForClaim := func(client ctrlruntimeclient.WithWatch, ns, name string, claim *hivev1.ClusterClaim, timeout time.Duration) error {
```

Comment:
Why does this need to be a closure as opposed to a method on `*clusterClaimStep`?

Reply:
Updated.